src/rpdk/core/project.py

```diff
@@ -217,15 +221,37 @@ def generate(self):
         # generate template for IAM role assumed by cloudformation
         # to provision resources if schema has handlers defined
         if "handlers" in self.schema:
+            handlers = self.schema["handlers"]
             template = self.env.get_template("resource-role.yml")
             permission = "Allow"
             path = self.root / ROLE_TEMPLATE_FILENAME
             LOG.debug("Writing Resource Role CloudFormation template: %s", path)
             actions = {
                 action
-                for handler in self.schema["handlers"].values()
+                for handler in handlers.values()
                 for action in handler.get("permissions", [])
             }
+
+            # calculate IAM role max session timeout based on highest handler timeout
+            # with some buffer (70 seconds per minute)
+            cud_handlers = [
+                handlers.get(handler, {})
+                for handler in ["create", "update", "delete"]
+                if handlers.get(handler, None)
+            ]
+            max_handler_timeout = max(
+                (
+                    handler.get("timeoutInMinutes", DEFAULT_ROLE_TIMEOUT_MINUTES)
+                    for handler in cud_handlers
+                ),
+                default=DEFAULT_ROLE_TIMEOUT_MINUTES,
+            )
+            # max role session timeout must be between 15 minutes and 12 hours
+            role_session_timeout = min(
+                MAX_ROLE_TIMEOUT_SECONDS,
+                max(MIN_ROLE_TIMEOUT_SECONDS, 70 * max_handler_timeout),
```

Comment:
Why it's `70` not `60`?

Reply:
for buffer time -- we have the exact same logic on CloudFormation side when setting the session duration for the role